dev/source/docs/building-px4-with-qtcreator.rst

```diff
@@ -0,0 +1,233 @@
+.. _building-px4-with-qtcreator:
+
+=====================================================================
+Building ArduPilot for Pixhawk/PX4 on Windows or Linux with QtCreator
+=====================================================================
+
+This article shows how you can set up Qt Creator for editing ArduPilot code
+and building for Pixhawk/PX4 targets on Windows and Linux.
+
+Preconditions for Linux
+=======================
+
+Follow the instructions in :ref:`Building for Pixhawk/PX4 on Linux with Make 
+<building-px4-for-linux-with-make>` to download the required source code 
+(*ardupilot*, *PX4Firmware* and *PX4NuttX*) and toolchain.
+
+Preconditions for Windows
+=========================
+
+Follow the instructions in :ref:`Pixhawk/PX4 on Windows with Make <building-px4-with-make>` 
+to download the required source code (*ardupilot*, *PX4Firmware* and *PX4NuttX*) and toolchain.
+
+Make sure you have no Cygwin installed (or have it at least out of the environment variables), 
+as this can get in the way of the PX4 toolchain.
+
+Install Qt Creator
+==================
+
+#. Download Qt and the Qt Creator IDE from the `Qt Creator website <https://www.qt.io/ide/>`__
+#. Follow the instructions to install the IDE on your platform.
+
+Run Qt Creator on Windows
+=========================
+
+We try to make sure that Qt Creator is running with all the necessary environment variables 
+that are needed to build an ArduPilot project.
+
+#. Go to your local PX4 toolchain directory and create a file called **qtcreator.sh**
+
+   .. code-block:: bash
+
+		cd /path-to-your-qt-creator-dir/bin
+		qtcreator.exe
+ 
+#. Go to the *toolchain\msys\1.0* subdirectory of the PX4 toolchain directory and
+   make a copy of the file **px4_console.bat** , called **px4_qt_creator.bat**.
+   Change this file in *:startsh* section, so that it becomes:
+
+   .. code-block:: bash
+
+		:startsh                             
+		if NOT EXIST %WD%sh.exe goto notfound
+		start %WD%sh --login -i -c creator.sh
+		exit   
+		
+#. Optional: create a Windows shortcut to the **px4_qt_creator.bat** for easy access.
+
+#. Start up Qt Creator by starting up **px4_qt_creator.bat**
+
+Run Qt Creator on Linux
+=======================
+
+If you installed the gcc-arm cross-compiler and made sure that the cross-compiler is in your
+path, then it suffices to simply start up Qt Creator.
+
+Create a project
+================
+
+#. Select File -> New File or Project
+
+#. Choose the "Import Project" template and from these templates "Import Existing Project". Then press "Next"
+
+   .. image:: ../images/QtCreator_ImportProject.png
+    :target: ../_images/QtCreator_ImportProject.png
```

Comment:
Does having a single space of indentation on the "target" work? Typically 3 spaces required for this.